### PR TITLE
topology: add numa distances

### PR DESCRIPTION
**Pull request by fromanirh**
Report distances between NUMA nodes as reported by sysfs.

This PR depends on https://github.com/jaypipes/ghw/pull/203 (whose commits are also added here to make the PR compilable) and also serves to demonstrate how to consume the snapshots programmatically, a feature added in https://github.com/jaypipes/ghw/pull/202.

With https://github.com/jaypipes/ghw/pull/203 merged, only https://github.com/jaypipes/ghw/commit/a2f2485401627ee302e8d846f1e75aa2d326db80 is relevant for this PR.

### CODE REVIEW

**Review by jaypipes — CHANGES_REQUESTED** [review]
Hi again @fromanirh, thank you so much for this! I'm really digging this new code.

I actually have only requested thing for you to fix up, which would be to add a line to the README docs for the `ghw.TopologyNode` struct:

https://github.com/jaypipes/ghw#topology

with a description of the new `Distances` field please.

Extra credit for a future PR might be to add to the example code in the README doc section above that shows the user how to, say, list the TopologyNodes "closest" to a particular node.

Other than that, this is awesomesauce.

Best,
-jay

File: pkg/topology/topology_linux.go [hunk 1]
```diff
@@ -60,7 +62,44 @@ func topologyNodes(ctx *context.Context) []*Node {
...
+		fmt.Sprintf("node%d", nodeID),
+		"distance",
+	)
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	items := strings.Split(strings.TrimSpace(string(data)), distanceSeparator)
```
[the remark this answers is not in the document]
Comment (fromanirh) [reply pkg/topology/topology_linux.go:95]: indeed, very cool! done

**Comment by fromanirh** [comment]
Updated as requested. The extra credit is a very nice idea, will work on that direction.

**Review by jaypipes — APPROVED** [review]
:+1: awesome work again, @fromanirh!

**Pull request merged**